train.py

```diff
@@ -0,0 +1,253 @@
+        model.load_state_dict(torch.load(f"models/{path}.pth"))
+
+    test_data = BagOfPitches(split="test", selected_composers=classnames)
+    test_dataloader = DataLoader(test_data, batch_size=cfg.hyperparameters.batch_size, shuffle=True)
+
+    true, pred = test_model(model, test_data=test_dataloader)
+    make_confusion_matrix(y_true=true, y_pred=pred, classes=classnames)
+    return validation_epoch(loader=test_dataloader, model=model, criterion=nn.CrossEntropyLoss())
+
+
+def one_run(cfg: DictConfig):
+    """
+    Function for testing
+    """
+    dataset = BagOfPitches(split="validation")
+    # get composers to classify against each other
+    count = dataset.df.groupby(["composer"]).size()
+    composers_with_most_data = count[count.values > 5].index.tolist()
+
+    run_experiment(cfg, classnames=composers_with_most_data)
+
+
+def composer_comparison_main(cfg: DictConfig):
+    """
+    Finds composers with at least 5 pieces in the validation split,
+    then trains model for each pair by calling run_experiment
+
+    Args:
+        cfg (DictConfig): DictConfig passed from hydra.main function containing hyperparameters and model specification.
+    """
+
+    composers_to_check = find_composers_to_check()
+    composers = [pair for pair in itertools.combinations(composers_to_check, r=2)]
+    print(f"pairs to check: {len(composers)}")
+    project = "MIDI-18-bag-of-pitches-pairs"
+    # container for trained models
+    models = []
+
+    # train model for each pair
+    for pair in composers:
+        print(f"{pair[0]} vs {pair[1]}")
+        model = run_experiment(cfg=cfg, project=project, classnames=pair)
+        models.append((model, pair))
+        first_composer = pair[0].replace(" ", "_").lower()
+        other_composer = pair[1].replace(" ", "_").lower()
+        path = f"models/{first_composer}-{other_composer}.pth"
+        if cfg.model.save_model == "y":
+            torch.save(model.state_dict(), path)
+
+
+def find_composers_to_check():
+    """
+    Finds well-represented composers in the dataset.
+
+    Returns:
+        list [str]: list of composers with at least 5 pieces in the validation split of the dataset.
+    """
+    # load data
+    dataset = BagOfPitches(split="validation")
+
+    # get composers with at least 5 pieces in the database
+    count = dataset.df.groupby(["composer"]).size()
+    composers_with_most_data = count[count.values > 5]
+    composers_to_check = composers_with_most_data.index.tolist()
+    return composers_to_check
+
+
+def initialize_wandb(cfg: DictConfig, project: str, classnames: list[str]):
+    # initialize experiment on WandB with unique run id
+    run_id = str(uuid.uuid1())[:8]
+    name = classnames[0].replace(" ", "_").lower()
+    for classname in classnames[1:]:
+        name += f"-{classname.replace(' ', '_').lower()}"
+    run = wandb.init(
+        project=project,
+        name=f"{name}-{run_id}",
+        config=OmegaConf.to_container(cfg, resolve=True),
+    )
```

Comment:
```suggestion
def initialize_wandb(cfg: DictConfig, classnames: list[str]):
    # initialize experiment on WandB with unique run id
    run_id = str(uuid.uuid1())[:8]
    name = classnames[0].replace(" ", "_").lower()
    for classname in classnames[1:]:
        name += f"-{classname.replace(' ', '_').lower()}"
    run = wandb.init(
        project=cfg.project,
        name=f"{name}-{run_id}",
        config=OmegaConf.to_container(cfg, resolve=True),
    )
```

Try to use the config for anything that feels hardcoded :)